### PR TITLE
[GAL-5013] [GAL-4899] Polish textarea for comment and post composer

### PR DESCRIPTION
**Pull request by jakzaizzat**
### Summary of Changes

- Character limit for the url in the comment/post composer will be divided by `10`
- Remove the character counter in web and mobile
- Auto expand input height on mobile

### Demo or Before/After Pics

**Character limit for the url**

https://github.com/gallery-so/gallery/assets/4480258/f665a097-8092-4b9f-810f-4b651fc64fee

**Auto increase height text input**

https://github.com/gallery-so/gallery/assets/4480258/19e57cd8-86ce-4ee9-ac07-a728892ec96e

**Remove the character counter in web**

https://github.com/gallery-so/gallery/assets/4480258/5dd192ee-c47b-42fe-88cd-2d9ceae7e576

**Remove the character counter in mobile**

https://github.com/gallery-so/gallery/assets/4480258/fcf0a9b2-3fe2-4c19-95b7-b4ac32a892c4

### Edge Cases

List any common edge cases that you have considered and tested.

### Testing Steps

Provide steps on how the reviewer can test the changes.

### Checklist

Please make sure to review and check all of the following:

- [x] I've tested the changes and all tests pass.
- [x] (if web) I've tested the changes on various desktop screen sizes to ensure responsiveness.
- [x] (if mobile) I've tested the changes on both light and dark modes.

### CODE REVIEW

File: apps/mobile/src/components/Post/PostInput.tsx [hunk 1]
```diff
@@ -34,67 +38,72 @@ export function PostInput({ value, onChange, tokenRef, mentions, onSelectionChan
...
+      >
+        <MentionTextInput
+          value={value}
+          onChangeText={onChange}
+          className="p-3 text-sm"
+          selectionColor={colorScheme === 'dark' ? colors.white : colors.black['800']}
+          placeholderTextColor={colorScheme === 'dark' ? colors.metal : colors.shadow}
+          multiline
+          onSelectionChange={(e: NativeSyntheticEvent<TextInputSelectionChangeEventData>) => {
+            onSelectionChange(e.nativeEvent.selection);
```
Comment (Robinnnnn) [inline apps/mobile/src/components/Post/PostInput.tsx:64]: let's make sure to wrap in `useCallback`

File: apps/mobile/src/components/Post/PostInput.tsx [hunk 2]
```diff
@@ -34,67 +38,72 @@ export function PostInput({ value, onChange, tokenRef, mentions, onSelectionChan
...
+        {isTextTooLong ? (
+          <Animated.View entering={transitionFadeIn} exiting={transitionFadeOut}>
+            <Typography
+              className="text-sm text-red"
+              font={{
+                family: 'ABCDiatype',
+                weight: 'Medium',
+              }}
+            >
+              Max text level reached
```
Comment (Robinnnnn) [inline apps/mobile/src/components/Post/PostInput.tsx:87]: length*, not level :)

you've been playing too much pokemon

File: packages/shared/src/utils/getRemaningCharacterCount.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+export const MAX_COMMENT_LENGTH = 300;
+export const MAX_POST_LENGTH = 600;
+
+export function getRemaningCharacterCount(message: string, MAX_TEXT_LENGTH: number) {
```
Comment (Robinnnnn) [inline packages/shared/src/utils/getRemaningCharacterCount.ts:4]: typo: Remaining*

Comment (Robinnnnn) [reply packages/shared/src/utils/getRemaningCharacterCount.ts:4]: can we also apply this to `CollectionCreateOrEditForm`? we want the collection descriptions on web to have a new MAX_LENGTH of 2400, and also not count urls.

Comment (Robinnnnn) [reply packages/shared/src/utils/getRemaningCharacterCount.ts:4]: finally, it'd be great to get chatGPT to write some tests for this :)

File: apps/mobile/src/components/Post/PostInput.tsx (hunk 1 above)
Comment (jakzaizzat) [reply apps/mobile/src/components/Post/PostInput.tsx]: Updated!

File: apps/mobile/src/components/Post/PostInput.tsx (hunk 2 above)
Comment (jakzaizzat) [reply apps/mobile/src/components/Post/PostInput.tsx]: lol 😂. Updated it!

File: packages/shared/src/utils/getRemaningCharacterCount.ts (hunk 3 above)
Comment (jakzaizzat) [reply packages/shared/src/utils/getRemaningCharacterCount.ts]: Updated the typo and added it to `CollectionCreateOrEditForm` too


https://github.com/gallery-so/gallery/assets/4480258/f6b4f737-c53c-417c-8989-5ac2d99d1514

Comment (jakzaizzat) [reply packages/shared/src/utils/getRemaningCharacterCount.ts]: Added test too!

**Review by Robinnnnn — APPROVED** [review]

**Pull request merged**